phpBB/includes/functions_upload.php

```diff
@@ -899,6 +909,18 @@ function remote_upload($upload_url, \phpbb\mimetype\guesser $mimetype_guesser =
 						$file = new fileerror($user->lang[$this->error_prefix . 'URL_NOT_FOUND']);
 						return $file;
 					}
+					else if (stripos($line, 'location: ') !== false)
+					{
+						//there is a redirect, follow up to 5
+						if ($redirect_count > 5)
```

Comment:
Missing the initial request. The initial request is 0, 1 means currently processing the first redirect. 5 means currently processing the 5th redirect. So it seems right to me